src/execution/experimental-executions.js

```diff
@@ -406,64 +560,65 @@ const createContributionPoPMetric = (measure, mdObj, measureIndex) => {
                 format
             }
         },
-        sort: getMetricSort(get(measure, 'sort'), true),
+        sort: getMeasureSorting(),
         meta: {
             measureIndex,
             isPoP: true
         }
-    }];
-
-    result.push(generated);
-
-    return result;
+    };
 };
 
-const categoryToElement = ({ category }) =>
-    ({ element: get(category, 'displayForm'), sort: get(category, 'sort') });
+const categoryToElement = (attributesMap, mdObj, context, category) => {
+    // for catalogue columns contain attribute uri, but for execution display form uri
+    const element = context === 'catalogue' ? getAttrUriFromMap(get(category, ['displayForm', 'uri']), attributesMap) : get(category, ['displayForm', 'uri']);
```

Comment:
[Q] When do we need all these conversions for `execution`?

Reply:
good point. After deprecation of `getDataForVis` only context for calling `mdToExecutionConfiguration` is getting catalogue items. No real execution anymore. Will clean the code